runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/SimpleStreamingWordCountTest.java

```diff
@@ -45,33 +49,44 @@
  */
 public class SimpleStreamingWordCountTest implements Serializable {
 
-  private static final String[] WORDS_ARRAY = {
-      "hi there", "hi", "hi sue bob", "hi sue", "", "bob hi"};
-  private static final List<Iterable<String>> WORDS_QUEUE =
-      Collections.<Iterable<String>>singletonList(Arrays.asList(WORDS_ARRAY));
-  private static final String[] EXPECTED_COUNTS = {"hi: 5", "there: 1", "sue: 2", "bob: 2"};
-
   @Rule
   public TemporaryFolder checkpointParentDir = new TemporaryFolder();
 
   @Rule
   public TestOptionsForStreaming commonOptions = new TestOptionsForStreaming();
 
+  private static final String[] WORDS = {"hi there", "hi", "hi sue bob", "hi sue", "", "bob hi"};
+
+  private static final List<Iterable<String>> MANY_WORDS =
+      Lists.<Iterable<String>>newArrayList(Arrays.asList(WORDS), Arrays.asList(WORDS));
+
+  private static final String[] EXPECTED_WORD_COUNTS = {"hi: 5", "there: 1", "sue: 2", "bob: 2"};
+
+  private static final Duration BATCH_INTERVAL_MILLIS = Duration.standardSeconds(1);
+
   @Test
   public void testRun() throws Exception {
+
     SparkPipelineOptions options = commonOptions.withTmpCheckpointDir(
         checkpointParentDir.newFolder(getClass().getSimpleName()));
 
-    Pipeline p = Pipeline.create(options);
-    PCollection<String> inputWords =
-        p.apply(CreateStream.fromQueue(WORDS_QUEUE)).setCoder(StringUtf8Coder.of());
-    PCollection<String> windowedWords = inputWords
-        .apply(Window.<String>into(FixedWindows.of(Duration.standardSeconds(1))));
-    PCollection<String> output = windowedWords.apply(new WordCount.CountWords())
-        .apply(MapElements.via(new WordCount.FormatAsTextFn()));
+    // override defaults
+    options.setBatchIntervalMillis(BATCH_INTERVAL_MILLIS.getMillis());
+    options.setTimeout(BATCH_INTERVAL_MILLIS.multipliedBy(3).getMillis());
```

Comment:
The fact that you mock a stream with 1 or 2 micro-batches is irrelevant to the Beam model (although it is a part of the upgrade to this test).
In terms of the model, you're using a `FixedWindow` of `2 seconds` and testing an implicit "processing time" trigger with "discarding pane" - simply because that's how Spark behaves.
It's kind of weird to test a `2 seconds` window with a `1 second` timeout, no ?